### DISCUSSION TITLE
Actualisation on PowerBI

### DISCUSSION THREAD

**Discussion by OrlandoPicqArch** [tmijail/Odoo-Power-BI-Connector#8] [Q&A] [answered]
Hello Mijail,

First of all, thanks very much for this connector between Odoo and PowerBI.
It works very well on PowerBI desktop.

I'd like your advice how to actualise the data from the web. Like if I want to planify an automatic update every day.
At the moment, I can only actualise from the desktop app.

See below the screenshot of the problem :
![image](https://user-images.githubusercontent.com/72440439/149350920-3007c771-5d97-4c52-a39e-6ecbc7ba84f6.png)

Would appreciate your help.
Kr,
Orlando

**Comment by tmijail** [comment]
Could you translate the message?

**Reply by OrlandoPicqArch** [reply]
Sorry, everything on the screenshot is in French.
The message in yellow reads :

You cannot schedule a refresh for this dataset because the following data sources do not support refresh at this time:
Data source for Query1
Explore data sources

**Reply by tmijail** [reply]
Hmmm... That's weird. 
Do you have a [data gateway](https://docs.microsoft.com/en-us/power-bi/connect-data/service-gateway-onprem) set up?
Does the "Base Odoo" dataset use any source other than Odoo?

**Reply by OrlandoPicqArch** [reply]
No, I haven't set up any data gateway. Will read this and try to set one up.
Second question, no there is only one source : Odoo

**Reply by tmijail** [reply]
That's probably it, then. 
After you set up the gateway, you should be able to assign the gateway to the dataset

![imagen](https://user-images.githubusercontent.com/6920322/149358094-a10b88cf-50a9-42ee-adf9-80a9b3b18e91.png)

**Reply by OrlandoPicqArch** [reply]
Ok I understand now. But if I set up a gateway, it means that my computer has to be online in order to refresh the database.
I mean, the actualisation won't run automatically at 5am if my computer is not connected to the internet

**Reply by OrlandoPicqArch** [reply]
Now, I've set up the gateway on my computer but I still don't have the gateway line on the configuration.
In fact, I have difficulty on the configuration on powerbi :

see screenshot : maybe I haven't filled the form correctly
![image](https://user-images.githubusercontent.com/72440439/149360531-1fb15b28-0f04-4665-b45c-5c62fd779972.png)

**Reply by OrlandoPicqArch** [reply]
Do I have to set up the standard Gateway or the personal mode ?
Yet, I've tried with both but it doesn't seem to appear here :
![image](https://user-images.githubusercontent.com/72440439/149365361-854da3f5-6c9b-419b-9d03-fbe8ed4356ad.png)

**Comment by tmijail** [comment]
> Ok I understand now. But if I set up a gateway, it means that my computer has to be online in order to refresh the database.
I mean, the actualisation won't run automatically at 5am if my computer is not connected to the internet

Yes, that is correct. Your computer will download the data from Odoo and then upload it to PBI. I think it's not possible for PBI to connect directly to a non SQL source through a custom connector. If it's possible, Microsoft hasn't documented how.

> Now, I've set up the gateway on my computer but I still don't have the gateway line on the configuration.
In fact, I have difficulty on the configuration on powerbi :

Did you install the `data gateway` or the `data gateway (personal mode)`? (see [types of gateways](https://docs.microsoft.com/en-us/power-bi/connect-data/service-gateway-onprem#types-of-gateways)). Both work fine, but the configuration is a little different. The personal mode gateway is simpler.

> see screenshot : maybe I haven't filled the form correctly

That datasource is not using the Odoo connector, it's trying to use the PostgreSQL connector. Servers hosted on [odoo.com](odoo.com) don't allow you to connect directly to the PostgresSQL database, so that's probably the cause of the error. For a Odoo datasource, you should see something like this:

![imagen](https://user-images.githubusercontent.com/6920322/149379599-69e4ad1e-09bd-45f5-97ab-1e6c6af343b8.png)

**Reply by OrlandoPicqArch** [reply]
Hello,
Thanks for your answer.

Unfortunately, I don't have the 'Odoo' label on the second droplist on 'Type of database'
![image](https://user-images.githubusercontent.com/72440439/149467859-db1ac8b0-36f3-4389-a7d4-4c331a4c9ba4.png)

**Comment by OrlandoPicqArch** [comment]
Also, is it normal that I don't have the possibility to link the gateway on the settings ?
![image](https://user-images.githubusercontent.com/72440439/149488811-19cf8d8c-c33a-4e84-a809-e6eb3663604b.png)

In comparison, I had some previous connection with ODBC before that and I can see the possibility to link the gateway.
![image](https://user-images.githubusercontent.com/72440439/149489120-12972fec-9eec-4ded-b4f0-3e6dd4933dde.png)

**Comment by tmijail** [comment, marked as answer]
Ok, so your gateway is working for everything but the custom connector.

Does Odoo show up on the "Connectors" section of the On-premises data gateway?

![2022-01-14 09-03-40](https://user-images.githubusercontent.com/6920322/149513047-a671a1dc-99c9-44bf-8139-7ae12ef33eb5.gif)

If not, what does the "Load custom data connectors from folder" field say? Is there an `Odoo.mez` file in that folder? Could you send it to me?

If that field is empty or it's set to a folder that doesn't contain `Odoo.mez`, try setting it to the Power BI custom connectors folder `%USERPROFILE%\Documents\Power BI Desktop\Custom Connectors`

**Reply by OrlandoPicqArch** [reply]
Yet, after refreshing PowerBI, I can't configure the gateway with Odoo
![image](https://user-images.githubusercontent.com/72440439/149538441-847e1ace-7e14-4f3b-a28d-391b5853c649.png)

**Reply by OrlandoPicqArch** [reply]
See attached the Odoo.mez
I had to put it in a zip folder
[Odoo.zip](https://github.com/tmijail/Odoo-Power-BI-Connector/files/7871290/Odoo.zip)

**Reply by tmijail** [reply]
Try restarting the gateway. It's very finicky.

![imagen](https://user-images.githubusercontent.com/6920322/149539531-67d12a92-7db9-4d16-a4e3-20ca237e9d2d.png)

If that doesn't work, I'll do some testing with your `Odoo.mez` file after work.

**Reply by OrlandoPicqArch** [reply]
Yep, I've restarted it.
Still doesn't work

Do I have to complete something here ?
![image](https://user-images.githubusercontent.com/72440439/149542775-27932572-d758-4d15-b82a-1926413ead6c.png)

**Reply by tmijail** [reply]
I don't think so.

I have the first two checkboxes enabled.

![imagen](https://user-images.githubusercontent.com/6920322/149543782-bfb0ed0a-76c5-41c3-95ee-d5fd656f121b.png)

It's been a long time since I set my gateway up though, so I don't remember if those checkboxes were necessary. I'll try to set up a gateway from scratch to see what we missed.

**Reply by OrlandoPicqArch** [reply]
Yep, I can find Odoo now 👍  with the 2 checkboxes enabled. I can move forward with the rest and let you know

**Reply by tmijail** [reply]
Awesome!

**Reply by OrlandoPicqArch** [reply]
Gateway ok now
![image](https://user-images.githubusercontent.com/72440439/149545131-bb36c7ce-c54a-4262-aebc-cb5788bd8858.png)

**Reply by OrlandoPicqArch** [reply]
Now that the gateway is active, I still can't configure it with this connector 
![image](https://user-images.githubusercontent.com/72440439/149546385-88231b06-bd6a-4374-9f4d-153ce2b49d8d.png)

**Reply by OrlandoPicqArch** [reply]
So it comes back to this yellow message

"You cannot schedule a refresh for this dataset because the following data sources do not support refresh at this time:
Data source for Query1
Explore data sources"

Is it a problem with the connector?

**Reply by OrlandoPicqArch** [reply]
I confim it is very finicky because when I republished the report from the PowerBI desktop to the web portal, I can now see the "gateway connection" on the page

Now the problem is the gateway where it shows on the screenshot an 'incorrect configuration'
Even with the personal mode, it doesn't work

![image](https://user-images.githubusercontent.com/72440439/149551483-8ee0cdab-ab03-4634-adc0-5998a93da01b.png)

**Reply by OrlandoPicqArch** [reply]
I don't know why because on the gateways, it shows that it works
![image](https://user-images.githubusercontent.com/72440439/149551845-27ea9617-b418-4895-a6ab-9d500e3fde65.png)

**Reply by OrlandoPicqArch** [reply]
Which one do you use ?
![image](https://user-images.githubusercontent.com/72440439/149552395-17693549-ce1d-43d5-a741-10e5a7942fe5.png)

**Reply by OrlandoPicqArch** [reply]
I've put it to None
Also, the URL has to be like this : arch.odoo.com (so without any https://)

**Reply by OrlandoPicqArch** [reply]
Thank you very much for helping me on this. I don't know how to thank you.
Works very well
I've been able to update the database and schedule automated updates twice a day
![image](https://user-images.githubusercontent.com/72440439/149555618-f29c5652-0bbe-460f-9349-265a697c7c5c.png)

**Reply by tmijail** [reply]
That's good to hear!

**Comment by OrlandoPicqArch** [comment]
So yes I confirm the gateway couldn't find the connector : maybe because my documents are synchronised with OneDrive (C:\Users\OrlandoPicq\OneDrive - ARCH.DESIGN\Documents\Power BI Desktop\Custom Connectors)
So I paste the Odoo.mez on the initial path and the gataway finds the connector (C:\WINDOWS\ServiceProfiles\PBIEgwService\Documents\Power BI Desktop\Custom Connectors)
![image](https://user-images.githubusercontent.com/72440439/149537241-c6de609d-a8c0-483a-ab70-52946de07a43.png)
![image](https://user-images.githubusercontent.com/72440439/149537541-9d6b8e83-216a-43c0-8433-cabaa09131b3.png)
![image](https://user-images.githubusercontent.com/72440439/149538085-85cefd27-24b0-4849-a753-ae18b934fe1b.png)

Btw, what do you use to add a video or gif on the post ?

**Reply by tmijail** [reply]
> Btw, what do you use to add a video or gif on the post ?

I use OBS to record an .mp4, a site like https://ezgif.com to convert it to gif and then just drag and drop the file to the comment.
I just found out you can also drag and drop the mp4

https://user-images.githubusercontent.com/6920322/149538706-a232166c-ee6e-420c-90e4-8282c4f0ed60.mp4

**Reply by juliogarmar** [reply]
Hello there.

First of all thanks for this connector, it is extremely useful for Odoo users as me.

I have follow some guides to make use of your custom connector trough a gateway installed in my computer and everything semmed ok until the last step. Trying to connect to the Odoo database in PowerQuery (running on PowerBi Service) I get this error:

![image](https://user-images.githubusercontent.com/82391655/152899486-ca483475-5d22-4231-b11a-248c5c7b78b6.png)

Apparently it is replacing the slashes in the "https://..." URL to "https:\/\/..." Maybe that is the source of the error, but I am not sure. I have tried many different combinations of slashes, without any luck.

Any idea about how to solve it?

Thank you in advance!

**Reply by OrlandoPicqArch** [reply]
It works for me but I don't use the https...
![image](https://user-images.githubusercontent.com/72440439/152947888-143a1708-48c2-4c7f-9df9-520f4f9e8569.png)

**Reply by juliogarmar** [reply]
But you are not running it as a DataFlow from PowerBi Service. Try, you will see it does not work.
